services/QuillCMS/lib/modules/response_aggregator.rb

```diff
@@ -32,6 +42,17 @@ def optimality_counts_of_question(question_uid)
     }
   end
 
+  def question_dashboard_data(question_uid)
+    total_number_of_responses = Response.where(question_uid: question_uid).sum('count')
+    common_unmatched = Response.where(question_uid: question_uid, parent_id: nil, optimal: nil).where('count >= 10').sum('count')
+    algorithm_matched = Response.where(question_uid: question_uid, author: DASHBOARD_ALGOS).sum('count')
+
+    {
+      percent_common_unmatched: (common_unmatched.to_f / (total_number_of_responses.nonzero? || 1)) * 100,
+      percent_specified_algos: (algorithm_matched.to_f / (total_number_of_responses.nonzero? || 1)) * 100
```

Comment:
Interesting - `nonzero?` is the only Rails API I've encountered that breaks the convention that methods ending in ? return booleans.

Reply:
`File#size?` is another gotcha.

Reply:
I think what's most confusing is `zero?` works as expected.